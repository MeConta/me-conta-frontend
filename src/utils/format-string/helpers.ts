// Rever tipagem
export const formatPhoneNumber = (value: string): any =>
  value
    .replace(/\D/g, '')
    .replace(/(\d{2})(\d)/, '($1) $2')
    .replace(/(\s\d{1})/, '$1 ')
    .replace(/(\d{4})(\d)/, '$1-$2')
    .replace(/(-\d{4})(\d+?)$/, '$1')

export const removePhoneMask = (value: string) => value.replace(/\D/g, '')
