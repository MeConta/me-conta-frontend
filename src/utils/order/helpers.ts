import { VolunteerAvailableSlot } from '../../services/volunteers-service/volunteer-service'

function convertSlotsIntoString(dateSlot: VolunteerAvailableSlot) {
  return new Date(dateSlot.inicio).toLocaleTimeString([], {
    hour: '2-digit',
    minute: '2-digit',
    hour12: false
  })
}

export const sortSlotsByTime = (dateSlot: VolunteerAvailableSlot[]) => {
  return dateSlot.sort((a, b) => {
    const x = convertSlotsIntoString(a)
    const y = convertSlotsIntoString(b)
    return x == y ? 0 : x > y ? 1 : -1
  })
}
