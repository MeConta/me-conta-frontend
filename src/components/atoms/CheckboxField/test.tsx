import { render, screen, waitFor, act } from '../../../utils/tests/helpers'
import userEvent from '@testing-library/user-event'

import { CheckboxField } from '.'

describe('<CheckboxField/>', () => {
  it('should not change value with text input', async () => {
    render(<CheckboxField data-testid="checkbox" label="checkbox" />)
    expect(screen.getByTestId('checkbox')).toBeInTheDocument()
  })

  it('should change value when clicked', () => {
    render(
      <CheckboxField data-testid="checkbox" label="checkbox" name="checkbox" />
    )
    act(() => {
      userEvent.click(screen.getByTestId('checkbox'))
    })
    expect(screen.getByLabelText(/checkbox/i)).toBeChecked()
  })

  it('should disable input with disabled propriety', async () => {
    const mockChange = jest.fn()
    render(
      <CheckboxField
        name="checkbox"
        data-testid="checkbox"
        disabled
        onChange={mockChange}
      />
    )
    const input = screen.getByTestId('checkbox')
    expect(input).toBeDisabled()

    userEvent.click(input)
    await waitFor(() => {
      expect(mockChange).not.toHaveBeenCalled()
    })
  })

  it('should render error the input', () => {
    const errorMessage = 'Error message'
    const { container } = render(
      <CheckboxField
        name="checkbox"
        data-testid="checkbox"
        error={errorMessage}
      />
    )
    expect(screen.getByText(errorMessage)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
