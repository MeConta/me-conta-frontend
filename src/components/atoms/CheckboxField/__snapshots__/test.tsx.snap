// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxField/> should render error the input 1`] = `
.c0 {
  margin-bottom: 1.6rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  vertical-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.7rem;
  padding: 0.8rem 1.6rem;
}

.c3 {
  color: #DE3163;
  font-size: 1rem;
}

.c2 {
  height: 2 rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <input
      class="c2"
      data-testid="checkbox"
      name="checkbox"
      type="checkbox"
      value=""
    />
  </div>
  <p
    class="c3"
  >
     
    Error message
     
  </p>
</div>
`;
